usr/src/uts/armv8/rpi4/io/bcm2711_pcie/bcm2711_pcie.c

```diff
@@ -29,8 +31,22 @@
 #include <sys/pci_cfgacc.h>
 #include <sys/pcie_impl.h>
 
-#define	BCM2711_DEV_CFG_DATA	0x8000 /* (read/write) config space data */
-#define	BCM2711_DEV_CFG_INDEX	0x9000 /* (write) config space index  */
+#define	BCM2711_REG_CONTROLLER_HW_REV	0x406c
+#define	BCM2711_REG_BRIDGE_CTRL		0x9210
+#define	BCM2711_BRIDGE_DISABLE_FLAG	0x1
+#define	BCM2711_BRIDGE_RESET_FLAG	0x2
+#define	BCM2711_REG_PCIE_HARD_DEBUG	0x4204
+#define	BCM2711_REG_BRIDGE_STATE	0x4068
+#define	BCM2711_REG_BRIDGE_LINK_STATE	0x00bc
+
+#define	BCM2711_DEV_CFG_DATA		0x8000 /* (read/write) config space data */
+#define	BCM2711_DEV_CFG_INDEX		0x9000 /* (write) config space index  */
+
+/* XXXPCI: should we define these "globally" */
```

Comment:
These are in sys/pcie_impl.h as `PCIE_LINK_SPEED_2_5` etc.

Reply:
removed the link speed message. If we want to add something we should probably use the generic framework.